src/client/language/languageConfiguration.ts

```diff
@@ -6,15 +6,15 @@ import { IndentAction, languages } from 'vscode';
 import { PYTHON_LANGUAGE } from '../common/constants';
 
 export const MULTILINE_SEPARATOR_INDENT_REGEX = /^(?!\s+\\)[^#\n]+\\$/;
+export const INCREASE_INDENT_REGEX = /^\s*(?:async|class|def|elif|else|except|finally|for|if|try|while|with)\b.*:\s*(#.*)?$/;
+export const DECREASE_INDENT_REGEX = /^\s*(?:elif|else|except|finally)\b.*:\s*(#.*)?$/;
+export const OUTDENT_SINGLE_KEYWORD_REGEX = /^\s*(break|continue|pass|raise)\b.*(#.*)?$/;
+export const OUTDENT_RETURN_REGEX = /^\s*(return)\b([^\[\(\{})]|([\[\(\{].*[\]\)\}]))*(#.*)?$/;
```

Comment:
Do characters in a character group (`[...]`) really have to be escaped?  Also, using `(?:` helps make grouping more visually distinctive.

```suggestion
export const OUTDENT_RETURN_REGEX = /^\s*(?:return)\b(?:[^[({})]|(?:[[({].*[\])}]))*(#.*)?$/;
```

Also, yikes that regex is hard to read!  (not your fault)  This would definitely benefit from a comment explaining what is going on.  Or maybe we just don't worry about the brackets?

Reply:
> yikes that regex is hard to read!

☝️  my reaction when a regex has more than 8 characters 🙃  Anyway, that's a good point, the backslash wasn't necessary for all but one `]`. I'll add a comment as well 👍